ngsPETSc/utils/firedrake/meshes.py

```diff
@@ -12,6 +12,12 @@
 import numpy as np
 from petsc4py import PETSc
 
+try:
```

Comment:
We can also add a hard dependency on `scipy`.

Reply:
Okay!